### PR TITLE
Added C++ implementation of bisection method in bisection branch

### LINKED ISSUE #1914: Bisection method in cpp
I would like to add a solution in cpp to implement root finding method i.e The Bisection method.Please assign me this issue

### PR DESCRIPTION
**Pull request by Nikhil-Patro**
Fixes #1914 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added a C++ implementation to the bisection method that finds the root of a given polynomial

#### Languages Used:
- C++
#### Files Added:
- Bisection_Method.cpp

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by Nikhil-Patro** [comment]
@asha15  There ,done :) I hope everything is correct now.Please review it.

File: Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp [hunk 1]
```diff
@@ -0,0 +1,83 @@
+#include<iostream>
+#include<cmath>
+using namespace std;
+#define error 0.0001 //approximation limit
```
Comment (asha15) [inline Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp]: Take the comment into previous line

File: Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+//function of the polynomial is inputted here say f(x) = x*x - 2*x + 1
```
Comment (asha15) [inline Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp]: keep a line space before this

File: Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+float f(float x)
+{
+    return x*x - 4*x + 3;
```
Comment (asha15) [inline Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp]: keep space before and after *

File: Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+}
```
Comment (asha15) [inline Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp]: keep a line space after this function

File: Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp [hunk 5]
```diff
@@ -0,0 +1,83 @@
+#include<iostream>
+#include<cmath>
+using namespace std;
+#define error 0.0001 //approximation limit
+//function of the polynomial is inputted here say f(x) = x*x - 2*x + 1
+float f(float x)
```
Comment (asha15) [inline Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp]: define a meaningful name for the function

File: Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+{
+    return x*x - 4*x + 3;
+}
+float mid(float q,float w)
```
Comment (asha15) [inline Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp]: keep a space after comma

File: Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+{
+    return (q + w)/2;
```
Comment (asha15) [inline Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp]: keep space before and after /

File: Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+}
```
Comment (asha15) [inline Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp]: keep a line space after this

File: Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+int main()
+{
+    int i;
+    float lower_bound,upper_bound;
```
Comment (asha15) [inline Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp]: keep a space after comma

File: Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+    cout<<"For the interval (a,b) where the root may lie : "<<endl;
+    cout<<"Enter the value of a :- "<<endl;
+    cin>>lower_bound;
+    cout<<"Enter the value of b :- "<<endl;
+    cin>>upper_bound;
+    if(f(lower_bound)*f(upper_bound) > 0)
+    {
+        cout<<"Invalid interval"<<endl;
```
Comment (asha15) [inline Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp]: keep space before and after >>,<< and *

File: Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+    }
+    else
+    {
+        while(abs(f(mid(lower_bound,upper_bound))) > error) // repeat until the required accuracy is obtained
```
Comment (asha15) [inline Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp]: keep a space after comma

File: Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+        {
+            cout<<"\nIteration_no = "<<i<<" ,First_no = "<<lower_bound<<" ,Second_no = "<<upper_bound<<" middle = "<<mid(lower_bound,upper_bound)<<" ,f(middle) = "<<f(mid(lower_bound,upper_bound))<<" ,f(first_no)*f(middle) = "<<f(lower_bound)*f(mid(lower_bound,upper_bound))<<endl;
```
Comment (asha15) [inline Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp]: break this into two lines

File: Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+            if(f(lower_bound)*f(mid(lower_bound,upper_bound)) < 0) //sign changes in the first interval => root lies in the 1st interval
```
Comment (asha15) [inline Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp]: take the commend into previous line 
keep space after comma

File: Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+            {
+                upper_bound=mid(lower_bound,upper_bound);
```
Comment (asha15) [inline Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp]: keep space after comma
keep space before and after =

File: Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+            }
+            else
+            {
+                lower_bound=mid(lower_bound,upper_bound);
```
Comment (asha15) [inline Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp]: keep space after comma
keep space before and after =

File: Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+
+Iteration_no = 7 ,First_no = 2.99062 ,Second_no = 3.00625 middle = 2.99844 ,f(middle) = -0.00312233 ,f(first_no)*f(middle) = 5.82703e-005
+
+Iteration_no = 8 ,First_no = 2.99844 ,Second_no = 3.00625 middle = 3.00234 ,f(middle) = 0.00469303 ,f(first_no)*f(middle) = -1.46532e-005
+
+Iteration_no = 9 ,First_no = 2.99844 ,Second_no = 3.00234 middle = 3.00039 ,f(middle) = 0.000781059 ,f(first_no)*f(middle) = -2.43872e-006
+
+Iteration_no = 10 ,First_no = 2.99844 ,Second_no = 3.00039 middle = 2.99941 ,f(middle) = -0.00117207 ,f(first_no)*f(middle) = 3.65958e-006
+
+Iteration_no = 11 ,First_no = 2.99941 ,Second_no = 3.00039 middle = 2.9999 ,f(middle) = -0.000195503 ,f(first_no)*f(middle) = 2.29143e-007
```
Comment (asha15) [inline Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp]: break this into two lines

File: Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp [hunk 17]
```diff
@@ -0,0 +1,83 @@
...
+
+Iteration_no = 12 ,First_no = 2.9999 ,Second_no = 3.00039 middle = 3.00015 ,f(middle) = 0.000292778 ,f(first_no)*f(middle) = -5.7239e-008
+
+Iteration_no = 13 ,First_no = 2.9999 ,Second_no = 3.00015 middle = 3.00002 ,f(middle) = 4.86374e-005 ,f(first_no)*f(middle) = -9.50877e-009
```
Comment (asha15) [inline Algo_Ds_Notes/Bisection_Method/Bisection_Method.cpp]: break this into two lines 
The input and output needs to be more clear

**Comment by Nikhil-Patro** [comment]
I have made all the suggested changes . Please review @asha15

File: Bisection_Method/Bisection_Method.cpp [hunk 18]
```diff
@@ -0,0 +1,105 @@
+#include<iostream>
+#include<cmath>
+using namespace std;
+// approximation limit
+#define error 0.0001
+
+//function of the polynomial is inputted here say f(x) = x*x - 2*x + 1
```
Comment (asha15) [inline Bisection_Method/Bisection_Method.cpp]: keep space around *

File: Bisection_Method/Bisection_Method.cpp [hunk 19]
```diff
@@ -0,0 +1,105 @@
...
+float polynomial(float x)
+{
+    return x * x - 4 * x + 3;
+}
+
+float mid(float q, float w)
+{
+    return (q + w)/2;
```
Comment (asha15) [inline Bisection_Method/Bisection_Method.cpp]: keep space around /

File: Bisection_Method/Bisection_Method.cpp [hunk 20]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+int main()
+{
+    int i;
+    float lower_bound,upper_bound,a,b;
```
Comment (asha15) [inline Bisection_Method/Bisection_Method.cpp]: keep space after comma

File: Bisection_Method/Bisection_Method.cpp [hunk 21]
```diff
@@ -0,0 +1,105 @@
...
+    cout<<"For the interval (a,b) where the root may lie : "<<endl;
```
Comment (asha15) [inline Bisection_Method/Bisection_Method.cpp]: keep after comma
keep space around << and >>

File: Bisection_Method/Bisection_Method.cpp [hunk 22]
```diff
@@ -0,0 +1,105 @@
...
+    cout<<"Enter the value of a :- "<<endl;
+    cin>>lower_bound;
+    a = lower_bound;
+    cout<<"Enter the value of b :- "<<endl;
+    cin>>upper_bound;
+    b = upper_bound;
+    if(polynomial(lower_bound) * polynomial(upper_bound) > 0)
+    {
+        cout<<"Invalid interval"<<endl;
```
Comment (asha15) [inline Bisection_Method/Bisection_Method.cpp]: keep space around << and >>

File: Bisection_Method/Bisection_Method.cpp [hunk 23]
```diff
@@ -0,0 +1,105 @@
...
+    }
+    else
+    {   // repeat until the required accuracy is obtained
```
Comment (asha15) [inline Bisection_Method/Bisection_Method.cpp]: take the commend into a new line

File: Bisection_Method/Bisection_Method.cpp [hunk 24]
```diff
@@ -0,0 +1,105 @@
...
+        while(abs(polynomial(mid(lower_bound, upper_bound))) > error)
+        {
+            cout<<"\nIteration_no = "<<i<<" ,First_no = "<<lower_bound<<" ,Second_no = "<<upper_bound<<" middle = "<<mid(lower_bound, upper_bound)<<" ,f(middle) = "<<polynomial(mid(lower_bound, upper_bound))<<" ,f(first_no)*f(middle) = "<<polynomial(lower_bound)*polynomial(mid(lower_bound, upper_bound))<<endl;
```
Comment (asha15) [inline Bisection_Method/Bisection_Method.cpp]: break this into two line. one line should contain maximum 80 characters 
keep space around << and >>

File: Bisection_Method/Bisection_Method.cpp [hunk 25]
```diff
@@ -0,0 +1,105 @@
...
+                upper_bound = mid(lower_bound, upper_bound);
+            }
+            else
+            {
+                lower_bound = mid(lower_bound, upper_bound);
+            }
+            i++;
+        }
+        cout<<"\nIteration_no = "<<i<<" ,First_no = "<<lower_bound<<" ,Second_no = "<<upper_bound<<" middle = "<<mid(lower_bound, upper_bound)<<" ,f(middle) = "<<polynomial(mid(lower_bound, upper_bound))<<" ,f(first_no)*f(middle) = "<<polynomial(lower_bound)*polynomial(mid(lower_bound, upper_bound))<<endl;
+        cout<<"\nThe approximated root of the given polynomial within the specified interval ("<<a<<" , "<<b<<") = "<<mid(upper_bound, lower_bound)<<endl;
```
Comment (asha15) [inline Bisection_Method/Bisection_Method.cpp]: break this into two lines

**Comment by asha15** [comment]
squash your commit before pushing the changes

**Comment by Nikhil-Patro** [comment]
Ok,i messed up while squashing commits.So,i created a new PR .Please review that one.I am closing this.

**Pull request closed without merging**